src/router.ts

```diff
@@ -0,0 +1,514 @@
+  statusCode: number
+  headers?: ResponseHeaders
+  body: string
+  isBase64Encoded?: boolean
+}
+
+export interface CustomResponse extends Omit<Response, 'body'> {
+  [CUSTOM_RESPONSE]: boolean
+  statusCode: number
+  headers?: ResponseHeaders
+  isBase64Encoded?: boolean
+  body?: unknown
+}
+
+export interface RouteHandler<Event, Context> {
+  (event: Event, context: Context): Promise<BodyResponse | CustomResponse>
+}
+
+interface Route<Event, Context> {
+  method: Method
+  path: string
+  handler: RouteHandler<Event, Context>
+}
+
+export type Middleware<Event, Context> = (
+  event: Event,
+  context: Context,
+  requestPath?: string,
+  httpMethod?: string
+) => Promise<void>
+
+export interface CustomResponseFn {
+  (options: {
+    statusCode?: number
+    body?: unknown
+    headers?: ResponseHeaders
+    isBase64Encoded?: boolean
+  }): CustomResponse
+  setHeader: (header: string, value: string) => void
+}
+
+export interface HttpError extends Error {
+  statusCode: number
+  headers: ResponseHeaders
+}
+
+type ProxyEvent = APIGatewayProxyEvent | APIGatewayProxyEventV2
+
+export type AddRoute<Event extends ProxyEvent, Context> = (
+  path: string,
+  handler: RouteHandler<RouterEvent<Event>, RouterContext<Context>>
+) => void
+
+export type RouterEvent<Event extends ProxyEvent> = Omit<Event, 'body'> & {
+  body?: string | null | Record<string, unknown>
+  rawBody?: string | null
+  rawHeaders: RequestHeaders
+}
+export type RouterContext<Context> = Context & {
+  includeTraceId?: boolean
+  traceId?: string
+  response: CustomResponseFn
+}
+export interface LambdaRouter<Event extends ProxyEvent, Context> {
+  get: AddRoute<Event, Context>
+  post: AddRoute<Event, Context>
+  delete: AddRoute<Event, Context>
+  put: AddRoute<Event, Context>
+  patch: AddRoute<Event, Context>
+  formatError: (handler: ErrorFormatter) => void
+  unknown: (
+    handler: (
+      event: RouterEvent<Event>,
+      context: RouterContext<Context>,
+      path: string | undefined,
+      httpMethod: string | undefined,
+      routes: Array<{ path: string; method: string }>
+    ) => Promise<BodyResponse | CustomResponse> | BodyResponse | CustomResponse
+  ) => void
+  route: (
+    event: Event,
+    context: Context,
+    requestPath?: string,
+    httpMethod?: string
+  ) => Promise<RouterResponse>
+  /** add middleware that receives the event & context before they are sent to route handler. Modifications are passed through */
+  beforeRoute: (handler: Middleware<RouterEvent<Event>, RouterContext<Context>>) => void
+}
+
+export interface RouterResponse {
+  endpoint?: string
+  uri: string
+  isOk: boolean
+  response: Response
+}
+
+export interface RouterProps {
+  logger?: any
+  extractPathParameters?: boolean
+  trimTrailingSlash?: boolean
+  includeTraceId?: boolean
+  includeErrorStack?: boolean
+  cors?: boolean | string
+  parseBody?: boolean
+  assumeJson?: boolean
+  decodeEvent?: boolean
+  normalizeHeaders?: boolean
+}
```

Comment:
`trimTrailingSlash` and `assumeJson` are missing in the readme's api section

Reply:
done